src/Storages/MergeTree/MergeTreeReaderWide.cpp

```diff
@@ -261,14 +279,7 @@ void MergeTreeReaderWide::readData(
     const auto & name = name_and_type.name;
     auto serialization = data_part->getSerialization(name_and_type);
 
-    if (deserialize_binary_bulk_state_map.count(name) == 0)
-    {
-        deserialize_settings.getter = [&](const ISerialization::SubstreamPath & substream_path)
-        {
-            return getStream(/* seek_to_start = */true, substream_path, streams, name_and_type, from_mark, /* seek_to_mark = */false, current_task_last_mark, cache);
-        };
-        serialization->deserializeBinaryBulkStatePrefix(deserialize_settings, deserialize_binary_bulk_state_map[name]);
-    }
+    deserializePrefix(serialization, name_and_type, current_task_last_mark, cache);
```

Comment:
shouldn't we rewind back? // like it was before https://github.com/ClickHouse/ClickHouse/blob/4b8e90338d867899ccc055d240a6b7554f24d7a7/src/Storages/MergeTree/MergeTreeReaderWide.cpp#L194

Reply:
and reset `deserialize_settings.continue_reading`?